### ISSUE TITLE
Why tf.losses.softmax_cross_entropy doesn't have "dim" (axis) argument?

### ISSUE THREAD

**Issue by iteal** [tensorflow/tensorflow#20866]
Hi!
The function "tf.losses.softmax_cross_entropy" calls "nn.softmax_cross_entropy_with_logits_v2" which has a dim argument with a default value of -1.
Can we add the same "dim" or "axis" parameter to the calling function " tf.losses.softmax_cross_entropy", so that we can choose which axis to apply the softmax  ?
Thanks!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by laetitiaoist** [comment]
Have I written custom code N/A
OS Platform and Distribution N/A
TensorFlow installed from N/A
TensorFlow version N/A
Bazel version N/A
CUDA/cuDNN version N/A
GPU model and memory N/A
Exact command to reproduce N/A

**Comment by tatianashp** [comment]
We currently don't have anybody working on this API. It would be great if you could help us by implementing the change that you propose and submitting a PR. Let us know if you need further clarification. Thanks!
/cc @alextp @ispirmustafa

**Comment by Siddhant085** [comment]
I think its a simple change. I am new to this repository and I would like to work on this issue.

**Comment by tanzhenyu** [comment]
Just FYI there is already such implementation on keras categorical_crossentropy:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/backend.py#L3461

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by jaketae** [comment]
I'm wondering if this issue should be closed given that `tf.losses` is deprecated in TensorFlow 2.X, replaced instead by `tf.keras.losses` as pointed out [here](https://github.com/tensorflow/tensorflow/pull/21996#issuecomment-486023491).

**Comment by tilakrayal** [comment]
@iteal,
CategoricalCrossentropy computes the loss between labels and predictions.

Use crossentropy loss function when there are two or more label classes. We expect labels to be provided in a one_hot representation. 
If you want to provide labels as integers, please use SparseCategoricalCrossentropy loss. There should be classes floating point values per feature.

```
tf.keras.losses.CategoricalCrossentropy(
    from_logits=False,
    label_smoothing=0.0,
    axis=-1,
    reduction="auto",
    name="categorical_crossentropy",
)
```
Also please take a look at this [comment](https://github.com/tensorflow/tensorflow/issues/20866#issuecomment-407537773) from the developer which delivers the information. Thank you!

**Comment by iteal** [comment]
As this issue was 4 years ago, I have since moved on and not using TensorFlow anymore, so you can close this ticket.

**Comment by tilakrayal** [comment]
@tatianashp,
Could you please feel free to move this issue to closed status. Thank you!
